core/src/main/java/hudson/cli/BuildCommand.java

```diff
@@ -134,15 +140,15 @@ protected int run() throws Exception {
             a = new ParametersAction(values);
         }
 
-        if (checkSCM) {
-            if (job.poll(new StreamTaskListener(stdout, getClientCharset())).change == Change.NONE) {
+        if (checkSCM && job instanceof AbstractProject<?, ?>) {
+            if (((AbstractProject<?, ?>)job).poll(new StreamTaskListener(stdout, getClientCharset())).change == Change.NONE) {
                 return 0;
             }
         }
 
         if (!job.isBuildable()) {
             String msg = Messages.BuildCommand_CLICause_CannotBuildUnknownReasons(job.getFullDisplayName());
-            if (job.isDisabled()) {
+            if (job instanceof AbstractProject<?, ?> && ((AbstractProject<?, ?>)job).isDisabled()) {
```

Comment:
JENKINS-27299 for reference, but anyway this only comes down to the choice of a message.